### PR TITLE
Fixes hd5 weight preprocessing / loading

### PR DESCRIPTION
**Pull request by kentsommer**
This is a patch for the issue describe in #5885

### CODE REVIEW

File: keras/engine/topology.py [hunk 1]
```diff
@@ -2886,7 +2886,7 @@ def load_weights_from_hdf5_group(f, layers):
     for k, name in enumerate(layer_names):
         g = f[name]
         weight_names = [n.decode('utf8') for n in g.attrs['weight_names']]
-        weight_values = [g[weight_name] for weight_name in weight_names]
+        weight_values = np.array([g[weight_name] for weight_name in weight_names])
```
Comment (kentsommer) [inline keras/engine/topology.py]: @fchollet 

Yes that is correct, sorry about that!

**Comment by fchollet** [comment]
I don't think this is the right fix. If the problem is with `convert_kernel`, then it should be fixed there. And casting should only be done following a type check, to avoid unnecessary casting.

**Comment by kentsommer** [comment]
I'm not sure the issue should necessarily be considered to be with ```convert_kernel```. Previously, the datatype passed to ```conver_kernel``` was ALWAYS a numpy array, however, the data being passed to it now is sometimes an hd5 dataset which obviously breaks things. 

So the data cleansing should (in my mind) happen somewhere upstream from ```convert_kernel```.

**Comment by fchollet** [comment]
You should definitely move the cast to `convert_kernel`. Please also add a unit test for `convert_kernel`.

**Comment by fchollet** [comment]
Closing due to no activity in a while.

**Pull request closed without merging**